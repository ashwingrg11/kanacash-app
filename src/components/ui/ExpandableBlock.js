import React from 'react';
import {View, StyleSheet, TouchableOpacity} from 'react-native';
import theme from '~/components/theme/Style';
import {MediumText, RegularText, SemiBoldText} from '~/components/ui/Text';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/Ionicons';

export default function ExpandableBlock({
  left,
  primary,
  title,
  caption,
  leftContent,
  rightContent,
  onPress,
  backgroundColor,
  containerStyle,
  content,
  style,
}) {
  const invert = primary ? primary : backgroundColor;
  return (
    <View style={[styles.container(primary, backgroundColor), style && style]}>
      <TouchableOpacity onPress={onPress}>
        <View style={[styles.topWrapper, containerStyle && containerStyle]}>
          {leftContent}
          <View style={styles.textWrapper}>
            {caption !== null ? (
              <View style={styles.titleWrapper}>
                <MediumText text={title} invert={invert} />
                {title === 'Home Delivery' && (
                  <View style={styles.homeDeliveryWrapper}>
                    <Icon
                      name="md-warning"
                      size={20}
                      style={styles.iconStyle}
                    />
                    <RegularText text="Yerevan only" />
                  </View>
                )}
              </View>
            ) : (
              <SemiBoldText text={title} invert={invert} />
            )}
            {caption && <RegularText text={caption} invert={invert} />}
          </View>
          {rightContent && (
            <View style={styles.rightContentStyle}>{rightContent}</View>
          )}
        </View>
      </TouchableOpacity>
      {content && content}
    </View>
  );
}

ExpandableBlock.defaultProps = {
  primary: false,
};

ExpandableBlock.propTypes = {
  leftContent: PropTypes.element,
  rightContent: PropTypes.element,
  onPress: PropTypes.func,
};

const styles = StyleSheet.create({
  container: (primary, backgroundColor) => ({
    borderRadius: theme.defaultRadius,
    backgroundColor: primary
      ? theme.primaryColor
      : backgroundColor
      ? backgroundColor
      : theme.white,
    paddingHorizontal: 10,
    paddingVertical: 12,
    marginVertical: 5,
    borderColor: '#E5E5E5',
    borderWidth: 1,
  }),
  topWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  textWrapper: {
    flex: 4,
    marginHorizontal: 10,
  },
  rightContentStyle: {
    alignItems: 'flex-end',
    flex: 1,
  },
  titleWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  homeDeliveryWrapper: {
    paddingHorizontal: 4,
    flexDirection: 'row',
    backgroundColor: '#FFFF99',
    marginLeft: 10,
    alignItems: 'center',
  },
  iconStyle: {
    color: '#999900',
    marginRight: 10,
  },
});
