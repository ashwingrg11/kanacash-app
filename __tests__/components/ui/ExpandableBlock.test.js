import 'react-native';
import React from 'react';

import ExpandableBlock from '~/components/ui/ExpandableBlock';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<ExpandableBlock/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ExpandableBlock backgroundColor="red" title="title"
    caption="caption"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
