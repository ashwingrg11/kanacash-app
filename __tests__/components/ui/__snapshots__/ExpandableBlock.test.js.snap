// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandableBlock/> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "red",
        "borderColor": "#E5E5E5",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginVertical": 5,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 4,
            "marginHorizontal": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            invert="red"
            style={
              Array [
                Object {
                  "color": "#1F1F1F",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "lineHeight": 22,
                },
                Object {
                  "color": "#fff",
                },
                undefined,
              ]
            }
            text="title"
          >
            title
          </Text>
        </View>
        <Text
          invert="red"
          style={
            Array [
              Object {
                "color": "#5C5C5C",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "color": "#fff",
              },
              undefined,
            ]
          }
          text="caption"
        >
          caption
        </Text>
      </View>
    </View>
  </View>
</View>
`;
